lib/Optimizer/GraphOptimizer/GraphOptimizer.cpp

```diff
@@ -2838,6 +2838,9 @@ bool OptimizeConversions::run(Function *F, const CompilationContext &cctx) {
       if (auto *BN = llvm::dyn_cast<Constant>(CN->getInput())) {
         auto newConst =
             convertConstant(*F->getParent(), *BN, CN->getResult().getType());
+        if (newConst == NodeValue()) {
+          continue;
+        }
         LOG_ASSERT(newConst) << "Constant conversion failed.";
```

Comment:
I think this line should have been triggering in this case. I'm wondering if it should have been `newConst.getNode()` here instead. Regardless, it should be removed since it will never be reached.

Reply:
Yes, you are correct this Int32-Int64 should not throw this error. I was looking at an older commit where Int2ITy-Int64ITy case was not covered. So now I don't have the test case. 
But won't it be better to not optimize the node and move ahead instead of giving Log Assert? In the cases of plain conversions, this case might occur.

Reply:
@garg1himanshu Yes, please go ahead and remove the `LOG_ASSERT()` and we can land this.

Reply:
I have made the requested changes.